dropwizard/service/src/test/java/org/apache/polaris/service/dropwizard/it/DropwizardServerManager.java

```diff
@@ -0,0 +1,108 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.dropwizard.it;
+
+import io.dropwizard.testing.ConfigOverride;
+import io.dropwizard.testing.ResourceHelpers;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.polaris.service.dropwizard.PolarisApplication;
+import org.apache.polaris.service.dropwizard.auth.TokenUtils;
+import org.apache.polaris.service.dropwizard.config.PolarisApplicationConfig;
+import org.apache.polaris.service.it.env.AuthToken;
+import org.apache.polaris.service.it.env.ClientCredentials;
+import org.apache.polaris.service.it.env.Server;
+import org.apache.polaris.service.it.ext.PolarisServerManager;
+
+public class DropwizardServerManager implements PolarisServerManager {
+  public static final String TEST_REALM = "POLARIS";
+
+  @Override
+  public String realm() {
+    return TEST_REALM;
+  }
+
+  @Override
+  public Server startServer(Map<String, String> featureConfig) {
+    return new Holder(featureConfig);
+  }
+
+  private static class Holder implements Server {
+    private final DropwizardAppExtension<PolarisApplicationConfig> ext;
+    private final AuthToken token;
+
+    public Holder(Map<String, String> featureConfig) {
+      Map<String, String> config = new HashMap<>();
+      featureConfig.forEach((k, v) -> config.put("featureConfiguration." + k, v));
+      // Bind to random port to support parallelism
+      config.put("server.applicationConnectors[0].port", "0");
+      config.put("server.adminConnectors[0].port", "0");
+
+      ConfigOverride[] overrides =
+          config.entrySet().stream()
+              .map((e) -> ConfigOverride.config(e.getKey(), e.getValue()))
+              .toList()
+              .toArray(new ConfigOverride[0]);
+      ext =
+          new DropwizardAppExtension<>(
+              PolarisApplication.class,
+              ResourceHelpers.resourceFilePath("polaris-server-integrationtest.yml"),
+              overrides);
+
+      try {
+        ext.before();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+
+      token = new AuthToken(getToken(), "root");
+    }
+
+    private String getToken() {
+      return TokenUtils.getTokenFromSecrets(
+          ext.client(),
+          baseUri().toString(),
+          adminCredentials().clientId(),
+          adminCredentials().clientSecret(),
+          TEST_REALM);
+    }
+
+    @Override
+    public URI baseUri() {
+      return URI.create(String.format("http://localhost:%d", ext.getLocalPort()));
+    }
+
+    @Override
+    public AuthToken adminToken() {
+      return token;
+    }
+
+    @Override
+    public ClientCredentials adminCredentials() {
+      return new ClientCredentials("test-admin", "test-secret");
```

Comment:
The old `TestEnv` extension used to append a "test ID" to the generated credentials afaict.

Reply:
I'm not sure that the purpose of that append was, TBH :sweat_smile: I believe having one admin user per realm (as in this case) is sufficient.

Currently, there's only one realm for Server Manager instance, so other implementations can chose to use unique admin user names, if they prefer.